dowhy/causal_refuters/sensitivity_analysis.py

```diff
@@ -0,0 +1,449 @@
+import statsmodels.formula.api as smfa
+import numpy as np
+from scipy.stats import t
+import pandas as pd
+import statsmodels.api as sma
+import logging
+import matplotlib.pyplot as plt
+
+
+class Sensitivity_analysis:
+    """
+    Class to perform sensitivity analysis
+    :param formula: The formula specifying the model
+    :param data: Pandas dataframe
+    :param treatment_name: name of treatment
+    :param q: percentage reduction for robustness value
+    :param standard_error: standard error in regression
+    :param estimate: estimate of regression
+    :param degree_of_freedom: degree of freedom of error in regression
+    :param h0: hypothesis
+    :param stats: dictionary for sensitivity statistics
+    :param benchmark_covariates: names of variables for benchmark bounding
+    :param kd: strength of association between benchmark and treatment variable to test with benchmark bounding
+    :param ky: strength of association between benchmark and outcome variable to test with benchmark bounding
+    :param r2dz_x: partial R^2  of putative unobserved confounder "z" with treatment "d", observed covariates "x" partialed out
+    :param r2yz_dx: partial R^2  of putative unobserved confounder "z" with outcome "y", observed covariates "x" and treatment "d" partialed out
+    :param r2dxj_x: partial R^2 of covariate Xj with treatment "d", covariates "x" excluding Xj are partialed out
+    :param r2yxj_dx:  partial R^2 of covariate Xj with outcome "y", covariates "x" excluding Xj and treatment "d" are partialed out
+    :param bounds_results: dataframe containing information about bounds and bias adjusted terms
+    :param stats: dictionary containing information like robustness value, partial R^2, estimate, standard error , degree of freedom, partial f^2, t-statistic
+    """
+    formula = None

```

Comment:
we do not need the formula specifying the model. That information is already captured in `estimand` that is supplied to refute_estimate. So no need to add it as a parameter. If needed, use the info from `identified_estimand`